components/definitions/definitions.js

```diff
@@ -0,0 +1,158 @@
+import domready from '../../assets/js/domready';
+
+export const classAnchor = 'definition__anchor';
+export const classDialog = 'popup';
+export const classDialogButton = 'popup__button';
+export const classDialogDocument = 'popup__document';
+
+export const idDialogMessage = 'd-message';
+
+export const attrResponse = 'data-popup-response';
+export const attrDataDialogCall = 'data-popup-call';
+export const attrDataTitle = 'data-title';
+
+export default function() {
+  return definitionDialog();
+}
+
+export function definitionDialog() {
+  anchorsToButton();
+  buttonClick();
+}
+
+// Replace anchor links with buttons
+export function anchorsToButton() {
+  let anchors = document.getElementsByClassName(classAnchor);
+  for (let anchor of anchors) {
+    // Replace <a> tag with <button> tag.
+    anchor.outerHTML = anchor.outerHTML
+      .replace(/<a/g, '<button')
+      .replace(/<\/a>/g, '</button>');
+  }
+}
+
+// Open popup
+export function buttonClick() {
+  const buttons = document.getElementsByClassName(classAnchor);
+  for (let button of buttons) {
+    button.onclick = function(e) {
+      e.preventDefault();
+
+      // define the popup element
+      let popup = document.getElementsByClassName(classDialog)[0];
+
+      // record the trigger element
+      let trigger = button.getAttribute('id')
+        ? button.getAttribute('id')
+        : 'trigger';
+
+      if (popup.getAttribute('open') === 'true') {
+        closeDialog(buttons, trigger, popup);
+      }
+
+      openDialog(buttons, button, popup, trigger);
+    };
+  }
+}
+
+// Open Dialog
+export function openDialog(buttons, button, popup, trigger) {
+  // retrieve custom close button wording, if any
+  let closeText = button.getAttribute(attrResponse)
+    ? button.getAttribute(attrResponse)
+    : 'close';
+
+  button.setAttribute('disabled', 'true');
+
+  // open popup and add roles
+  popup.setAttribute('tabindex', '0');
+  popup.setAttribute('open', 'true');
+  popup.setAttribute('role', 'alertpopup');
```

Comment:
There isn't an `alertpopup` role, I assume some find and replace broke this. Regardless, this isn't an `alertdialog` either:

> The difference with regular dialogs is that the alertdialog role should only used when an alert, error, or warning occurs. In other words, when a dialog's information and controls require the user's immediate attention alertdialog should be used instead of dialog.

From [using the 'alertdialog' role](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_alertdialog_role).

Arguably, this could be a [dialog](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_dialog_role) though. However, I don't think it should "trap" the users focus inside the dialog, they should be able to continue focusing on other elements on the page (in order) without dismissing the dialog. It definitely shouldn't be modal, either.